src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ColorSchemesPageViewModel.h"
+#include "ColorSchemesPageViewModel.g.cpp"
+
+#include <LibraryResources.h>
+#include "..\WinRTUtils\inc\Utils.h"
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    ColorSchemesPageViewModel::ColorSchemesPageViewModel(const Model::CascadiaSettings& settings) :
+        _settings{ settings }
+    {
+        _MakeColorSchemeVMsHelper();
+        CurrentScheme(_AllColorSchemes.GetAt(0));
+    }
+
+    void ColorSchemesPageViewModel::UpdateSettings(const Model::CascadiaSettings& settings)
+    {
+        _settings = settings;
+        _MakeColorSchemeVMsHelper();
+    }
+
+    void ColorSchemesPageViewModel::RequestSetCurrentScheme(Editor::ColorSchemeViewModel scheme)
+    {
+        CurrentScheme(scheme);
+
+        // Update the state of the page
+        _PropertyChangedHandlers(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L"CanDeleteCurrentScheme" });
+    }
+
+    bool ColorSchemesPageViewModel::RequestRenameCurrentScheme(winrt::hstring newName)
+    {
+        // check if different name is already in use
+        const auto oldName{ CurrentScheme().Name() };
+        if (newName != oldName && _settings.GlobalSettings().ColorSchemes().HasKey(newName))
+        {
+            return false;
+        }
+        else
+        {
+            // update the settings model
+            CurrentScheme().Name(newName);
+            _settings.GlobalSettings().RemoveColorScheme(oldName);
+            _settings.GlobalSettings().AddColorScheme(CurrentScheme().SettingsModelObject());
+            _settings.UpdateColorSchemeReferences(oldName, newName);
+            return true;
+        }
+    }
+
+    void ColorSchemesPageViewModel::RequestDeleteCurrentScheme()

```

Comment:
I still like having the `Request` prefix because the idea is that the VM has the final say - for now the implementation is such that the VM always allows it, but maybe in the future the VM would want to check for certain things first before performing the action (and if we get to that point we probably would want to update the return value to be a boolean, like in `RequestRename`, but that's a later thing).

Reply:
Also this is consistent with what we have in `AddProfile.h` and `AddProfile.cpp`, note that the `RequestAddNew()` function there always adds a new profile even though its a 'request'